runtime/literal.go

```diff
@@ -297,22 +299,34 @@ func fixedPointLiteralValue(memoryGauge common.MemoryGauge, expression ast.Expre
 }
 
 func LiteralValue(inter *interpreter.Interpreter, expression ast.Expression, ty sema.Type) (cadence.Value, error) {
+
 	switch ty := ty.(type) {
 	case *sema.VariableSizedType:
 		expression, ok := expression.(*ast.ArrayExpression)
 		if !ok {
 			return nil, LiteralExpressionTypeError
 		}
-
-		return arrayLiteralValue(inter, expression.Values, ty.Type)
+		array, err := arrayLiteralValue(inter, expression.Values, ty.Type)
+		exported := ExportType(ty, map[sema.TypeID]cadence.Type{})
+		arrayType, ok := exported.(cadence.ArrayType)
+		if !ok {
+			return nil, LiteralExpressionTypeError
+		}
+		return array.WithType(cadence.NewMeteredVariableSizedArrayType(inter, arrayType.Element())), err
```

Comment:
Here and below: Maybe refactor the sub-expression into a variable, to make the code easier to read (length/nesting complexity)

```suggestion
		arrayCadenceType := cadence.NewMeteredVariableSizedArrayType(inter, arrayType.Element())
		return array.WithType(arrayCadenceType), err
```